### ISSUE TITLE
Error when applying Patch 0005-efi-chainloader-use-shim-to-load-and-verify-an-image.patch

### ISSUE THREAD

**Issue by txdowty** [Wind-River/meta-secure-core#37]
I am encountering this error when building target secure-core-image:

### error
```
ERROR: grub-efi-2.06-r0 do_patch: Applying patch '0005-efi-chainloader-use-shim-to-load-and-verify-an-image.patch' on target directory '/home/tdowty/amd/poky-amd-kirkstone/build-v3000-kirkstone/tmp/work/zen3-amd-linux/grub-efi/2.06-r0/grub-2.06'
CmdError('quilt --quiltrc /home/tdowty/amd/poky-amd-kirkstone/build-v3000-kirkstone/tmp/work/zen3-amd-linux/grub-efi/2.06-r0/recipe-sysroot-native/etc/quiltrc push', 0, 'stdout: Applying patch 0005-efi-chainloader-use-shim-to-load-and-verify-an-image.patch
patching file grub-core/loader/efi/chainloader.c
Hunk #1 FAILED at 760.
Hunk #2 succeeded at 892 (offset 6 lines).
1 out of 2 hunks FAILED -- rejects in file grub-core/loader/efi/chainloader.c
Patch 0005-efi-chainloader-use-shim-to-load-and-verify-an-image.patch does not apply (enforce with -f)
```

Here is my set up:
### fetch.sh
```
export YOCTO_BRANCH="kirkstone"

git clone --single-branch --branch "${YOCTO_BRANCH}" "https://git.yoctoproject.org/poky" "poky-amd-${YOCTO_BRANCH}"
pushd  poky-amd-${YOCTO_BRANCH}

git clone --single-branch --branch ${YOCTO_BRANCH} https://git.openembedded.org/meta-openembedded
git clone --single-branch --branch ${YOCTO_BRANCH} https://git.yoctoproject.org/meta-dpdk
git clone --single-branch --branch ${YOCTO_BRANCH} https://git.yoctoproject.org/meta-amd
git clone --single-branch --branch ${YOCTO_BRANCH} https://github.com/Wind-River/meta-secure-core.git

git checkout --quiet tags/yocto-4.0.5
pushd meta-openembedded

git checkout --quiet 50d4a8d2a983a68383ef1ffec2c8e21adf0c1a79
popd

pushd meta-dpdk
git checkout --quiet 0e62d02f2755fbbf7dfa6e243381377c0a1cd97c
popd

pushd meta-amd
git checkout --quiet tags/kirkstone-e3000-ga-202301
popd
```

### setup.sh
```
export YOCTO_BRANCH="kirkstone"
export MACHINE="v3000"
export DISTRO="poky-amd"

cd ${DISTRO}-${YOCTO_BRANCH}

source ./oe-init-build-env build-${MACHINE}-${YOCTO_BRANCH}

tee conf/auto.conf <<EOF
# DL_DIR ?= "\${TOPDIR}/../downloads"
SSTATE_DIR ?= "${HOME}/yp/sstate"

MACHINE = "${MACHINE}"
DISTRO = "poky-amd"

# secure boot
INITRAMFS_IMAGE = "secure-core-image-initramfs"
DISTRO_FEATURES_NATIVE:append = " systemd ima tpm tpm2 efi-secure-boot luks"
DISTRO_FEATURES:append = " systemd ima tpm tpm2 efi-secure-boot luks modsign"
MACHINE_FEATURES_NATIVE:append = " efi"
MACHINE_FEATURES:append = " efi"
PACKAGE_CLASSES:append = " package_rpm"
# this is added by amd bsp
PACKAGE_CLASSES:remove = " package_deb"
INHERIT += "sign_rpm_ext"
SECURE_CORE_IMAGE_EXTRA_INSTALL ?= "\
    packagegroup-efi-secure-boot \
    packagegroup-tpm \
    packagegroup-tpm2 \
    packagegroup-ima \
    packagegroup-luks \
"
DEBUG_FLAGS:forcevariable = ""
IMAGE_INSTALL:append = " kernel-image-bzimage"

# Uncomment this line to modify the root parameter in boot command line if the default one
# is not working for you. It is helpful when secure boot is enabled.
#BOOT_CMD_ROOT = "/dev/hda2"
EOF


bitbake-layers add-layer ../meta-openembedded/meta-oe
bitbake-layers add-layer ../meta-openembedded/meta-perl
bitbake-layers add-layer ../meta-openembedded/meta-python
bitbake-layers add-layer ../meta-openembedded/meta-networking
bitbake-layers add-layer ../meta-openembedded/meta-filesystems
bitbake-layers add-layer ../meta-dpdk
bitbake-layers add-layer ../meta-amd/meta-amd-distro
bitbake-layers add-layer ../meta-amd/meta-amd-bsp

bitbake-layers add-layer ../meta-secure-core/meta
bitbake-layers add-layer ../meta-secure-core/meta-signing-key
bitbake-layers add-layer ../meta-secure-core/meta-tpm
bitbake-layers add-layer ../meta-secure-core/meta-tpm2
bitbake-layers add-layer ../meta-secure-core/meta-efi-secure-boot
bitbake-layers add-layer ../meta-secure-core/meta-integrity
bitbake-layers add-layer ../meta-secure-core/meta-encrypted-storage

```

### build
`bitbake secure-core-image`

**Comment by yizhao1** [comment]
This is because you are not using the latest poky kirkstone branch. The meta-secure-core kirkstone branch is always based on the latest poky kirkstone branch. The latest tag in poky kirkstone branch is 4.0.16, but I see you are using 4.0.5. Several grub CVE fixes have been merged into the kirkstone branch since 4.0.5. So, we refreshed the meta-secure-core grub patches accordingly. I strongly recommend updating poky to latest 4.0.16 to resolve this issue.
